gql/client.py

```diff
@@ -271,8 +271,12 @@ async def __aenter__(self):
             self.session = AsyncClientSession(client=self)
 
         # Get schema from transport if needed
-        if self.fetch_schema_from_transport and not self.schema:
-            await self.session.fetch_schema()
+        try:
+            if self.fetch_schema_from_transport and not self.schema:
+                await self.session.fetch_schema()
+        except Exception:
```

Comment:
Please specify `(TransportError, GraphQLError)` instead of `Exception`

Reply:
Thanks for the suggestion. I actually wasn't sure what kind of exceptions could be thrown since it depends on the transport.

After thinking about it: In my example I used the AIOHTTPTransport, which throws a ClientError which inherits directly from Exception. This wouldn't be caught with your suggestion.

Reply:
You're perfectly right. I thought about fixing this kind of leaky abstractions by trying to catch all the exceptions of the transports and reraising them as TransportErrors but it can get tricky...

Reply:
@leszekhanusz I checked what kind of error is thrown and it depends on the transport. `HttpRequestTransport` throws an `ConnectionError` from the underlying urllib3 and the `AIOHttpTransport` throws a custom `ClientError`. All of those inherit from the base exception. Therefore, I see no way to narrow down the exception here.

Does it matter any way? We want to make sure that the transport is closed on any error. And then we re-raise the exception.